### ISSUE TITLE
import and re-export does not work

### ISSUE THREAD

**Issue by yiminghe** [Microsoft/TypeScript#9927]
**TypeScript Version:**  1.8.0

**Code**

```ts
import xx from 'xx';
export default xx;
export {yy} from 'yy';
```

**Expected behavior:**
generated code exports xx and yy respectively

**Actual behavior:** 
empty.

try https://www.typescriptlang.org/play/

but
```ts
import xx from 'xx';
export default xx + 1;
```
works

**Comment by mhegazy** [comment]
Are these inferaces or type aliases? Can you provide more details about your sample?

**Comment by yortus** [comment]
TypeScript *does* generate code that exports the values from `xx` and `yy` as you expect (although the exported names are `default` and `yy` the way you have written it). It just doesn't work on https://www.typescriptlang.org/play/ because the external modules 'xx' and 'yy' can't be found, so code generation fails (or partially fails).

If you actually create a little project with the following three files:
```ts
// file: index.ts
import xx from './xx'; // NB: changed to relative path
export default xx;
export {yy} from './yy'; // NB: changed to relative path

// file: xx.ts
export default 42;

// file: yy.ts
export var yy;
```

And then compile using `tsc  index xx yy`, you should find it creates the following `index.js`:

```js
"use strict";
var xx_1 = require('./xx');
exports.__esModule = true;
exports["default"] = xx_1["default"];
var yy_1 = require('./yy');
exports.yy = yy_1.yy;
```

**Comment by yiminghe** [comment]
why
```js
import xx from 'xx';
export default xx + 1;
```
works in https://www.typescriptlang.org/play/, but
```js
import xx from 'xx';
export default xx;
```
does not work?


`xx` and `yy` are not typescript modules and do not have d.ts, I expect correct output even `can not find module error`.

**Comment by yortus** [comment]
@yiminghe *neither* of those examples works in the TypeScript playground - they both have the same error - `"Cannot find module 'xx'.`

![ts9927](https://cloud.githubusercontent.com/assets/1252361/17127458/5a0a3746-5338-11e6-9b0b-6ebba6c4c007.png)

If there is any error in the compilation, then there are no guarantees about what is emitted. The compiler tries to emit valid code, but depending on the type and location of the errors, the emitted code *may or may not* be valid or complete. You have found such an example where it *just happens* to be valid.

In short, if there are compilation errors, you should assume the output is invalid, since in such cases the compiler makes no guarantees about what gets emitted. If the output happens to be valid, that's just a bonus, not something you can rely on.

**Comment by DanielRosenwasser** [comment]
I was going to point you to #540, but I checked TypeScript 2.0-beta and it seemed to give you the emit you wanted. Feel free to reopen if that's not the case.

**Comment by yortus** [comment]
> I expect correct output even `can not find module error`.

@DanielRosenwasser since 'design limitation' was removed and 'fixed' added, can you clarify whether the OP's expectation is reasonable, or is it still advisable to treat emitted code as suspect when there are compiler errors (ie it's still a design limitation in general)?

**Comment by RyanCavanaugh** [comment]
Emitted code in the presence of syntax or resolution errors is not guaranteed to be correct as we have no idea what "correct" even *is* in that situation. This also applies to type errors when using `async`.
